amti/actions/review.py

```diff
@@ -80,29 +82,52 @@ def review_hit(
-                        if user_input in ['y', 'n']:
-                            approve = user_input == 'y'
+                    input_string = """Would you like to (a)ccept, (r)eject, (s)kip or
+                        (m)ark the assignment? [a/r/s/m]:"""
+                    while True:
+                        user_input = input(input_string).strip().lower()
+                        if user_input in ['a', 'r', 's']:
+                            break
+                        elif user_input == 'm':
+                            print('Assignment marked.')
+                            while True:
+                                user_input = input('Would you like to (a)ccept, (r)eject, or (s)kip this assignment?').strip().lower()
+                                if user_input in ['a', 'r', 's']:
+                                    break
+                            marked_reason = input('Reason for marking? (Can be left blank.)').strip()
+                            marked_assignments.append([assignment_id, user_input, marked_reason])
                         else:
-                            print('Please type either "y" or "n".')
+                            print('Please type either "a", "r", "s", or "m".')
 
-                    if approve:
+                    if user_input == 'a':  # accept
                         logger.info(f'Approving assignment (ID: {assignment_id}).')
                         client.approve_assignment(
                             AssignmentId=assignment_id,
                             OverrideRejection=False)
-                    else:
-                        logger.info(
-                            f'Did not approve assignment (ID: {assignment_id}).'
-                            f' Please make sure to manually reject it.')
+                        
+                    elif user_input == 'r':  # reject
+                        while True:
+                            user_input = input('Confirm rejection of this assignment [y/n]?').strip().lower()
+                            if user_input  == 'y':
+                                feedback = input('Assignment rejection feedback:').strip()
+                                client.reject_assignment(
+                                    AssignmentId=assignment_id,
+                                    RequesterFeedback=feedback)
+                                logger.info(f'Rejecting assignment (ID: {assignment_id}).')
+                            elif user_input == 'n':
+                                logger.info(f'Did not reject assignment (ID: {assignment_id}). Please review it again.')
+                                break
+                    else:  # skip
+                        logger.info(f'Skipping assignment (ID: {assignment_id}).')
+
+    return marked_assignments
 
 
 def review_batch(
         client,
         batch_dir,
-        approve_all):
+        approve_all,
+        mark_file=None):
```

Comment:
Renaming this to `mark_file_path` will make it clearer when reading the code that it's a path and not a File object.

Also, if we push the default logic into the script argument parsing (comment at amti/clis/review.py#L31-35), then we should remove the `None` default for this argument.

Reply:
Done